contracts/Market.sol

```diff
@@ -356,7 +358,44 @@ contract Market is IMarket, Ownable, ERC721 {
 
 		emit Settled(index, _bets[index].payout, result, recipient);
 
-		_burn(index);
+		_burn(uint256(index));
+	}
+
+	/* 
+	 * @dev Reverse a bet. Return the stake to the bettor and the loan to the vault
+	 * @param index The index of the bet
+	 * @param signature Signature from market owner
+	 */
+	function refund(uint64 index, SignatureLib.Signature calldata signature) external {
+		bytes16 REFUND_CMD = 0x726566756e6400000000000000000000; //Bytes for "refund"
+		bytes32 messageHash = keccak256(
+			abi.encodePacked(REFUND_CMD, address(this), index)
+		);
+
+		require(
+			isValidSignature(messageHash, signature) == true,
+			"refund: Invalid signature"
+		);
+		_refund(index);
+	}
+
+	function _refund(uint64 index) internal virtual {
+		Bet memory bet = _bets[index];
+		require(bet.settled == false, "refund: Bet has already settled");
+
+		bet.settled = true;
```

Comment:
maybe we should have an `enum BetStatus` that can be `INPLAY`, `SETTLED`, or `REFUNDED`?

Reply:
Interesting idea

Reply:
This has been discussed, it doesn't make sense.

1. The bet is always in play
2. Or it's paid out.